### PR TITLE
remove deprecated apsis

### PR DESCRIPTION
**Pull request by jmecosta**
handled #83

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
   <properties>
     <configuration>Debug</configuration>
     <jacoco.version>0.8.5</jacoco.version>
+    <java.version>11</java.version>
```
Comment (milbrandt) [inline pom.xml]: SonarQube scanners for 7.9 should still run on Java 8, only server requires already Java 11. Is Java 11 really required for the plugin?

I know that SonarCloud will require scanners with Java 11 in autum, but our plugin will not be installed there.

File: sonar-communityfsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSensor.java [hunk 2]
```diff
@@ -503,9 +503,6 @@ private void handleNonBlankCommentsMetricTag(InputFile inputFile) throws XMLStre
 
           if ("Line".equals(tagName)) {
             value++;
-
-            int line = Integer.parseInt(stream.getElementText());
-            fileLinesContext.setIntValue(CoreMetrics.COMMENT_LINES_DATA_KEY, line, 1);
```
Comment (milbrandt) [inline sonar-communityfsharp-plugin/src/main/java/org/sonar/plugins/fsharp/FSharpSensor.java:508]: ok, see https://jira.sonarsource.com/browse/SONAR-11064

File: sonar-communityfsharp-plugin/pom.xml [hunk 3]
```diff
@@ -44,7 +44,7 @@
         <groupId>org.sonarsource.sonarqube</groupId>
         <artifactId>sonar-plugin-api</artifactId>
         <!-- minimal version of SonarQube to support. -->
-        <version>7.9</version>
+        <version>7.9</version>
```
Comment (milbrandt) [inline sonar-communityfsharp-plugin/pom.xml]: I don't like these trailing whitespaces, but GitHub offers an option to suppress them to view ;)

**Comment by milbrandt** [comment]
Should be fine, have you tested the scanner on a SonarQube server? (as initialisation changed) Otherwise I can test it when I'm back in the company.

**Review by milbrandt — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jmecosta) [reply pom.xml]: our instance, after we updated a few plugins (sonar-cxx) is now required to have java 11. but im not sure if this has any effect on any thing... so i will remove it for now!

File: sonar-communityfsharp-plugin/pom.xml (hunk 3 above)
Comment (jmecosta) [reply sonar-communityfsharp-plugin/pom.xml]: i was opion the format option in the editor did clean that... apparently it did not

File: pom.xml (hunk 1 above)
Comment (milbrandt) [reply pom.xml]: Java 8 jars should also run on Java 11 - but not the other way round. Thanks

**Comment by jmecosta** [comment]
> y.

yep i will test later tonight! need to confirm if the java part will not extract the zip everytime it runs

**Comment by milbrandt** [comment]
> > y.
> 
> yep i will test later tonight! need to confirm if the java part will not extract the zip everytime it runs

Be careful and take care that it will extract every time the version of the plugin changes (up- and downgrade)!

**Review by milbrandt — APPROVED** [review]

**Comment by jmecosta** [comment]
good news, is that all is working on my side! bad news is that sonarqube 8.3 with msbuild runner we no longer get f# analysis (on this or master). they must have done something to mess this up... needs more testing! @milbrandt  what version are u in... working fine with 7.9 LTS

**Comment by milbrandt** [comment]
> good news, is that all is working on my side! bad news is that sonarqube 8.3 with msbuild runner we no longer get f# analysis (on this or master). they must have done something to mess this up... needs more testing! @milbrandt what version are u in... working fine with 7.9 LTS

We are still on 8.2. In the community forum 8.4 is announced for end of this/beginning of next week - at least a bug releated to IE11 is fixed there :)

Our projects using F# are all in the new Net.Sdk project file format

**Comment by jmecosta** [comment]
hum, do you mind testing on your side! just to confirm... i really doubt
8.3 would break this... but would be good to confirm this is working in
8.2... net sdk shouldnt be the problem, we are also with those. its more if
you are using msbuild sq runner or sonar cli (this ones everything works)

**Comment by milbrandt** [comment]
> hum, do you mind testing on your side! just to confirm... i really doubt 8.3 would break this... but would be good to confirm this is working in 8.2... net sdk shouldnt be the problem, we are also with those. its more if you are using msbuild sq runner or sonar cli (this ones everything works)


I did a test with 3.1 (build 376) and one of our F# projects (about 16k lines of F#): the SonarQube tasks worked fine and I got the same number of code smells like before. The bvuild uses Fake, the sonar tasks are in surrounding the yaml pipeline.

**Comment by jmecosta** [comment]
I suspect u are not using msbuild scanner under fake... this problem will
likely appear when u are mixing a project that contains c# and f# which
then u are forced to use msbuild scanner

**Comment by jmecosta** [comment]
But anyway this is not a regression from this pull request. But something
we. Eed to figure out later. Thanks for confirming

**Pull request merged**